src/chinadns.c

```diff
@@ -622,11 +696,9 @@ static void queue_add(id_addr_t id_addr) {
 
 static id_addr_t *queue_lookup(uint16_t id) {
   int i;
-  // TODO assign new id instead of using id from clients
-  for (i = 0; i < ID_ADDR_QUEUE_LEN; i++) {
-    if (id_addr_queue[i].id == id)
-      return id_addr_queue + i;
-  }
+  i = (uint8_t)(id & 0xff);
+  if (id_addr_queue[i].id == id)
+  	return id_addr_queue + i;
```

Comment:
`id` is picked by random rather than increment in glibc. If we use sharding here like this, we have a high collision rate here: 50% collision rate in 16 concurrent queries(compared to 256 concurrent queries before the change). This is not enough if we ChinaDNS is running on a LAN that have several users. We should use precise comparison instead. My suggestion is to undo the change here.